uwsdora.module

```diff
@@ -106,3 +106,52 @@ function uwsdora_views_api() {
     'path' => drupal_get_path('module', 'uwsdora') . '/views',
   );
 }
+
+/**
+ * Implements hook_convert_mods_to_citeproc_jsons_alter().
+ */
+function uwsdora_convert_mods_to_citeproc_jsons_alter(&$output, $mods) {
+  module_load_include('inc', 'citeproc', 'includes/converter');
+
+  $pages = $mods->xpath("//mods:mods[1]/mods:relatedItem[@type='host']/mods:part/mods:extent[@unit='pages']");
+
+  if (empty($pages)) {
+    $pages = $mods->xpath("//mods:mods[1]/mods:relatedItem[@type='host']/mods:part/mods:extent[@unit='page']");
+  }
+
+  $pages = (!empty($pages) ? $pages[0] : NULL);
+  if ($pages) {
+    add_mods_namespace($pages);
+
+    $total = $pages->xpath('mods:total');
+    $total = (!empty($total) ? (string) $total[0] : NULL);
+
+    $list = $pages->xpath('mods:list');
+    $list = (!empty($list) ? (string) $list[0] : NULL);
+
+    $start = $pages->xpath('mods:start');
+    $start = (!empty($start) ? (string) $start[0] : NULL);
+
+    $end = $pages->xpath('mods:end');
+    $end = (!empty($end) ? (string) $end[0] : NULL);
+
+    if ($total) {
+      $data = $total;
+    }
+    elseif ($list) {
+      $data = $list;
+    }
+    elseif ($start) {
+      $data = $start;
+
+      if ($end) {
+        $data .= "-" . $end;
+      }
+    }
+  }
+
+  // Alter readpaths and use newly processed page information.
+  $output['page'] = $data;
```

Comment:
Possible to get here without `$data` being defined, if nothing was found for `$pages` or no `total`, `list`, `start` or `end` was contained... Would/should throw warnings.